### ISSUE TITLE
ArrayIndexOutOfBoundsException in FrequencyData.maxFrequency()

### ISSUE THREAD

**Issue by pzweuj** [exomiser/Exomiser#565]
## Stack Trace

Here is the stack trace from the exception:

```plaintext
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 1
        at org.monarchinitiative.exomiser.core.model.frequency.FrequencyData.frequency(FrequencyData.java:169) ~[exomiser-core-14.0.0.jar:na]
        at org.monarchinitiative.exomiser.core.model.frequency.FrequencyData.maxFrequency(FrequencyData.java:327) ~[exomiser-core-14.0.0.jar:na]
```

## Relevant Code Snippets

An `ArrayIndexOutOfBoundsException` is thrown when trying to access the array with index -1.

The maxFrequency() method:

https://github.com/exomiser/Exomiser/blob/ffb2da103ca54a9de03ceecfc70c8ddfcebba3fd/exomiser-core/src/main/java/org/monarchinitiative/exomiser/core/model/frequency/FrequencyData.java#L326-L328

The frequency() method:
https://github.com/exomiser/Exomiser/blob/ffb2da103ca54a9de03ceecfc70c8ddfcebba3fd/exomiser-core/src/main/java/org/monarchinitiative/exomiser/core/model/frequency/FrequencyData.java#L168-L170

The findMaxSource() method:
https://github.com/exomiser/Exomiser/blob/ffb2da103ca54a9de03ceecfc70c8ddfcebba3fd/exomiser-core/src/main/java/org/monarchinitiative/exomiser/core/model/frequency/FrequencyData.java#L114-L125

## Environment Details
Exomiser version: 14.0.0
Java(TM) SE Runtime Environment (build 18.0.2.1+1-1)

## Suggested Improvement

```java
public Frequency maxFrequency() { 
    return (this.size == 0 || maxSource == -1) ? null : frequency(maxSource); 
} 
```

**Comment by pzweuj** [comment]
Further investigation revealed that the issue occurred because the **THOUSAND_GENOMES** annotation included a population frequency of 0.0000, while other databases were empty, causing **findMaxSource** to ultimately return -1.

Test Vcf
```
##fileformat=VCFv4.2
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	Test
11	34916640	.	T	C	28.3	PASS	.	GT:AD:DP:GQ:PL:VAF	0/1:58,36:94:28:28,0,45:0.382979
```

**Comment by julesjacobsen** [comment]
@pzweuj thanks for the feedback and example case. This is now fixed in [14.0.1](https://github.com/exomiser/Exomiser/releases/tag/14.0.1)
